src/main/java/com/epam/ta/reportportal/commons/querygen/Condition.java

```diff
@@ -181,7 +187,15 @@ public Object castValue(CriteriaHolder criteriaHolder, String value, ErrorType e
 	EXISTS("ex") {
 		@Override
 		public org.jooq.Condition toCondition(FilterCondition filter, CriteriaHolder criteriaHolder) {
-			return field(criteriaHolder.getAggregateCriteria()).isNotNull();
+			if (criteriaHolder.getQueryCriteria().equals(criteriaHolder.getAggregateCriteria())) {
+				return field(criteriaHolder.getAggregateCriteria()).isNotNull();
+			} else {
+				boolean exists = filter.isNegative() != BooleanUtils.toBoolean(filter.getValue());
+				Field<Integer> aggregatedCount = DSL.coalesce(arrayLength(arrayRemove(DSL.arrayAgg(field(criteriaHolder.getQueryCriteria())),
+						(String) null
+				)), 0);
+				return exists ? aggregatedCount.gt(0) : aggregatedCount.eq(0);
```

Comment:
Further the negative condition is applied as "not"
so it seems that if exists is 'true' further it will be changed to false.
check it out

Reply:
fixed